src/DeviceInterface/InterfacePrototype.js

```diff
@@ -294,14 +319,28 @@ class InterfacePrototype {
      * @param {string} type
      */
     async selectedDetail (data, type) {
-        this.activeFormSelectedDetail(data, type)
-    }
-
-    /**
-     * @param {IdentityObject|CreditCardObject|CredentialsObject|{email:string, id: string}} data
-     * @param {string} type
-     */
-    activeFormSelectedDetail (data, type) {
+        if (this.globalConfig.isTopFrame) {
+            /**
+             * This is overridden in the Overlay, so that instead of trying to fill a form
+             * with the selected credentials, we instead send a message to the native
+             * side. Once received, the native side will store that selection so that a
+             * subsequence call from main webpage can retrieve it via polling.
```

Comment:
So, we couldn't manage to get rid of the polling thing. Pity, is this also used in Windows?

Reply:
not yet on macOS - we should as a follow up. 

and no, Windows uses a request->response model with AbortControllers to handle lifecycle instead